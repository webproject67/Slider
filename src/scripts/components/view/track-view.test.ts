import TrackView from './track-view';
import Model from '../model/model';

const main = document.createElement('div');
main.id = 'banana';

const state = {
  flag: true,
  from: -10000,
  fromPercent: 0,
  max: 100,
  min: 0,
  progress: true,
  range: 'one',
  draft: 0,
  start: 1,
  configuring: 0,
  scale: true,
  step: 1,
  to: -10000,
  toPercent: 100,
  view: 'horizontal',
};

const model = new Model(main, state);
const trackView = new TrackView();

describe('snapshot', () => {
  test('renders correctly element1', () => {
    expect(new TrackView().getElement(model)).toMatchSnapshot();
  });

  test('renders correctly element2', () => {
    model.setValue(['range'], ['range']);
    expect(new TrackView().getElement(model)).toMatchSnapshot();
  });

  test('renders correctly element3', () => {
    model.setValue(['view'], ['vertical']);
    expect(new TrackView().getElement(model)).toMatchSnapshot();
  });

  test('renders correctly updated element4', () => {
    model.setValue(['view'], ['horizontal']);
    expect(new TrackView().getUpdatedElement(model)).toMatchSnapshot();
  });
});

describe('handleToggleMouseDown', () => {
  test('spyOn toggle mouseDown', () => {
    const somethingSpy = jest.spyOn(trackView, 'handleToggleMouseDown');
    let evt: any;
    trackView.handleToggleMouseDown(model, evt);
    expect(somethingSpy).toHaveBeenCalledTimes(1);
  });
});
