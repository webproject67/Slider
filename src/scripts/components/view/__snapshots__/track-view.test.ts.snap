// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly element1 1`] = `
<div
  class="slider__wrapper"
>
  
      
  <div
    class="slider__inner "
  >
    
        
    <div
      class="slider__scale "
    >
      
        
    </div>
    
        
        
    <div
      class="slider__toggle slider__toggle_maximum"
      style="left:100%"
    />
    
      
  </div>
  
    
</div>
`;

exports[`snapshot renders correctly element2 1`] = `
<div
  class="slider__wrapper"
>
  
      
  <div
    class="slider__inner "
  >
    
        
    <div
      class="slider__scale "
    >
      
        
    </div>
    
        
    <div
      class="slider__toggle slider__toggle_minimum"
      style="left:0%"
    />
    
        
    <div
      class="slider__toggle slider__toggle_maximum"
      style="left:100%"
    />
    
      
  </div>
  
    
</div>
`;

exports[`snapshot renders correctly element3 1`] = `
<div
  class="slider__wrapper"
>
  
      
  <div
    class="slider__inner slider__inner_with-height"
  >
    
        
    <div
      class="slider__scale slider__scale_with-height"
    >
      
        
    </div>
    
        
    <div
      class="slider__toggle slider__toggle_vertical-minimum"
      style="top:0%"
    />
    
        
    <div
      class="slider__toggle slider__toggle_vertical-maximum"
      style="top:100%"
    />
    
      
  </div>
  
    
</div>
`;

exports[`snapshot renders correctly updated element4 1`] = `
<div
  class="slider__wrapper"
>
  
      
  <div
    class="slider__inner "
  >
    
        
    <div
      class="slider__scale "
    >
      
        
    </div>
    
        
    <div
      class="slider__toggle slider__toggle_minimum"
      style="left:0%"
    />
    
        
    <div
      class="slider__toggle slider__toggle_maximum"
      style="left:100%"
    />
    
      
  </div>
  
    
</div>
`;
